### ISSUE TITLE
Connection tunnel collapse on failed handshakes

### ISSUE THREAD

**Issue by etherunit** [mysteriumnetwork/node#5492]
**Describe the bug**
Testing provider=0x04c1d46c4fb5ea330ed2e29b8c98d4feb9900ee7 (v1.17.4) and experiencing an issue with connection drops after activating the tunnel (even functioning for quite some time) on the client. These handshakes start failing some time after consecutive pings to the peer succeed.

```
DEBUG: [userspace-wg]2022/09/29 09:23:23 peer(382J…jTw4) - Sending handshake initiation
DEBUG: [userspace-wg]2022/09/29 09:23:23 peer(382J…jTw4) - Received handshake response
DEBUG: [userspace-wg]2022/09/29 09:23:23 peer(382J…jTw4) - Sending keepalive packet
```

`2022-09-29T09:17:36.148 DBG ../../../../source/core/connection/manager.go:924    > Received p2p keepalive ping with SessionID=40c6c3a5-3ee6-4954-a6b1-013b010c38a4 from 0x04C1d46C4fB5EA330Ed2e29b8C98d4FeB9900EE7`

But if I try to transfer some megabytes of data, the connection halts after some time and becomes unresponsive. I can also see in the node log output that there was indeed an initial handshake and that some particular amount of data was exchanged.

```
2022-09-29T09:17:56.723 DBG ../../../../source/session/pingpong/invoice_payer.go:154 > Invoice received: {65082771755404227419687425083504681141260329108303526807630132571049664205152 98177107327521 0 ff8faba5d8ce066aa9c2b1d5ef2282fd57dc5ef4818d9f508cdf221b69431193 0x04c1d46c4fb5ea330ed2e29b8c98d4feb9900ee7 137}
2022-09-29T09:17:56.724 DBG ../../../../source/session/pingpong/price_calculator.go:52 > Calculated price 2817839273013044. Time component: 8.4900457352641362762e+12, data component: 2.8093492272777808186e+15. Transferred: 21618916, duration: 35.047805344. Price 872070714470095/h, 139531314315215200/GiB
```

Then after some time it's finally collapses: 

```
DEBUG: [userspace-wg]2022/09/29 09:24:02 peer(382J…jTw4) - Retrying handshake because we stopped hearing back after 15 seconds
DEBUG: [userspace-wg]2022/09/29 09:24:02 peer(382J…jTw4) - Sending handshake initiation
2022-09-29T09:24:05.587 DBG ../../../../source/eventbus/event_bus.go:101         > Published topic="consumer_ping_p2p" event={SessionID:40c6c3a5-3ee6-4954-a6b1-013b010c38a4 Duration:74.404625ms}
DEBUG: [userspace-wg]2022/09/29 09:24:07 peer(382J…jTw4) - Handshake did not complete after 5 seconds, retrying (try 2)
DEBUG: [userspace-wg]2022/09/29 09:24:07 peer(382J…jTw4) - Sending handshake initiation
2022-09-29T09:24:10.660 DBG ../../../../source/eventbus/event_bus.go:101         > Published topic="consumer_ping_p2p" event={SessionID:40c6c3a5-3ee6-4954-a6b1-013b010c38a4 Duration:72.766312ms}
DEBUG: [userspace-wg]2022/09/29 09:24:12 peer(382J…jTw4) - Handshake did not complete after 5 seconds, retrying (try 3)
DEBUG: [userspace-wg]2022/09/29 09:24:12 peer(382J…jTw4) - Sending handshake initiation
```
`2022-09-29T09:24:26.831 ERR ../../../../source/core/quality/mysterium_morqa.go:193 > Failed to sent batch metrics request, 2 error="Post \"https://quality.mysterium.network/api/v3/batch\": net/http: request canceled while waiting for connection (Client.Timeout exceeded while awaiting headers)"`

The connection was not placed OnHold, therefore I had to manually disconnect/connect it.

Node logs: https://ctxt.io/2/AAAQiocQEA

**Comment by Zensey** [comment]
> 2022-09-29T09:24:26.831 ERR ../../../../source/core/quality/mysterium_morqa.go:193 > Failed to sent batch metrics request, 2 error="Post \"https://quality.mysterium.network/api/v3/batch\": net/http: request canceled while waiting for connection (Client.Timeout exceeded while awaiting headers)"

@soffokl 
could that requests to `quality.mysterium.network`  fail b/c host was unavailable on network ?

**Comment by Zensey** [comment]
@etherunit 
does this issue has a mass character ?
can we see changes, for ex. in grafana ?

**Comment by Nitrooo** [comment]
> @etherunit does this issue has a mass character ? can we see mass changes, for ex. in grafana ?

I'm not sure if I got your questions right, but I have personally collected evidence of hundreds of nodes failing in the very same way over the course of at least 24 hours and most probably of the last few weeks. So yes, this is having a massive impact on the whole network as potentially every node is currently affected.

**Comment by etherunit** [comment]
@Zensey requests are not failing and it keeps working for some time. It fails only once tunnel collapses. I included it in the description and you can easily check it by connecting yourself.

**Comment by etherunit** [comment]
Fixed in 1.18.3
